src/commands/extract.ts

```diff
@@ -0,0 +1,136 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { OpenDialogOptions, ProgressLocation, Uri, window, workspace } from "vscode";
+import * as Constants from '../constants';
+import { ApiTreeItem } from '../explorer/ApiTreeItem';
+import { ServiceTreeItem } from '../explorer/ServiceTreeItem';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { cpUtils } from '../utils/cpUtils';
+import { dotnetUtils } from '../utils/dotnetUtils';
+
+export async function extractService(node?: ServiceTreeItem): Promise<void> {
```

Comment:
90% of the code in this function and extractAPI function are the same. Refactor.